resourcetopo/dag_util.go

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2024 KusionStack Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package resourcetopo
+
+import (
+	"container/list"
+	"fmt"
+)
+
+// dagCheck implement a depth first algorithm to make sure the resource topology in this manager has no cycle
+func (m *manager) dagCheck() error {
```

Comment:
We can merge these util go files to only one, and move this method to manager.go